docs/configuration/config-file-reference.md

```diff
@@ -2114,6 +2114,15 @@ The `memcached_client_config` configures the client used to connect to Memcached
 # Use consistent hashing to distribute to memcache servers.
 # CLI flag: -<prefix>.memcached.consistent-hash
 [consistent_hash: <boolean> | default = false]
+
+# If we should do an SRV lookup. Else we do an host lookup. Port must be
+# provided if this is set to false.
+# CLI flag: -<prefix>.memcached.use-srv
+[use_srv: <boolean> | default = true]
```

Comment:
I've the feeling this may be difficult to extend over the time (ie. what when we'll add a 3rd service discovery mode? another boolean flag?).

From a config perspective, I believe we should have 1 single config option to specify the endpoints. An idea is to:
- Deprecate `hostname`
- Introduce `endpoints` which accepts a comma separated list of endpoints. Endpoints can be:
  - `hostname:port` (so no need for a dedicated `port`). This will be a straight DNS resolution (from record name to A/AAAA record, which is the IP address)
  - `dnssrvnoa+` prefix (ie. `dnssrvnoa+<hostname>`) which is the SRV NOA address resolution (which is that we're supporting right now). This is one of the prefixes supported by Thanos service discovery that will also be supported by the memcached client used by the blocks index cache (PR will land soon). Re-using the same Thanos' strategy would allow us to uniform service discovery across Cortex.

If you don't want to reinvent the wheel, you can just reuse the Thanos service discovery:
https://thanos.io/service-discovery.md/